pkg/net/listener.go

```diff
@@ -0,0 +1,82 @@
+package net
+
+import (
+	"crypto/tls"
+	"fmt"
+	"net"
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/alibaba/pouch/pkg/user"
+
+	"github.com/sirupsen/logrus"
+)
+
+func GetListener(addr string, tlsConfig *tls.Config) (net.Listener, error) {
+	addrParts := strings.SplitN(addr, "://", 2)
+	if len(addrParts) != 2 {
+		return nil, fmt.Errorf("invalid listening address: %s", addr)
```

Comment:
I think in a common function, we could give more details about the error message. How about adding like this:
```
fmt.Errorf("invalid listening address %s: address must be [protocol]://[address]", addr)
```